pygeos/decorators.py

```diff
@@ -83,3 +84,47 @@ def wrapped(*args, **kwargs):
                 arr.flags.writeable = old_flag
 
     return wrapped
+
+
+class ReturningProcess(multiprocessing.Process):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._pconn, self._cconn = multiprocessing.Pipe()
+        self._result = None
+
+    def run(self):
+        if not self._target:
+            return
+        try:
+            self._cconn.send(self._target(*self._args, **self._kwargs))
+        except Exception as e:
+            self._cconn.send(e)
+
+    @property
+    def result(self):
+        if self._result is None and self._pconn.poll():
+            self._result = self._pconn.recv()
+        return self._result
+
+
+def may_segfault(func):
```

Comment:
This is no longer used; now that the segfaults are either guarded against (in case of empty points) or are limited to Debug compiles of GEOS (hopefully rare for users).

It seems useful to archive this, but not sure that it should go into this PR.  If you keep it here, perhaps it would be helpful to add more comments to help guide devs in when to use it (i.e., this is a bit of a last resort for segfaults from GEOS that can't be otherwise guarded against)?

Reply:
I think it should not be in this repo if it isn't used. Instead, I created a gist: https://gist.github.com/caspervdw/10d9fd373a31642498cdea494d2f1bc4